tests/test_emulator.cpp

```diff
@@ -295,70 +305,64 @@ class EmulatorTest : public testing::Test {
             vcpu_original.gpr[REG_RCX] = test.in_src;
             vcpu_original.flags = test.in_flags;
             vcpu_expected = vcpu_original;
-            if (!readonly_dst)
-                vcpu_expected.gpr[REG_RDX] = test.out_dst;
+            vcpu_expected.gpr[REG_RDX] = test.out_dst;
             vcpu_expected.flags = test.out_flags;
             run(insn, vcpu_original, vcpu_expected);
         }
     }
 
-    template <int N>
-    void test_insn_rN_iN(const char* insn_name,
-                         const std::vector<test_alu_2op_t>& tests,
-                         bool readonly_dst = false) {
+    template <int N, int M=N>
+    void test_insn_rN_iM(const char* insn_name,
+                         const std::vector<test_alu_2op_t>& tests) {
         char insn[256];
         test_cpu_t vcpu_original;
         test_cpu_t vcpu_expected;
 
         // Run tests
         for (const auto& test : tests) {
             snprintf(insn, sizeof(insn), "%s %s, %d", insn_name,
-                gpr<N>(REG_RAX), (uint32_t)test.in_src);
+                gpr<N>(REG_RAX), (imm<M>)test.in_src);
```

Comment:
nit: We could take this chance to fix the indentation.

```suggestion
                     gpr<N>(REG_RAX), (imm<M>)test.in_src);
```